pkg/common/cns-lib/vsphere/virtualmachine_test.go

```diff
@@ -0,0 +1,204 @@
+package vsphere
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"testing"
+
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/simulator"
+	"github.com/vmware/govmomi/vapi/rest"
+	_ "github.com/vmware/govmomi/vapi/simulator"
+	"github.com/vmware/govmomi/vapi/tags"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+const (
+	regionCategoryName string = "k8s-region"
+	regionTagName      string = "region-1"
+	zoneCategoryName   string = "k8s-zone"
+	zoneTagName        string = "zone-A"
+)
+
+// CreateNewCategory creates a new category using tag manager
+func CreateNewCategory(ctx context.Context, categoryName string, cardinality string,
+	tagManager *tags.Manager) (string, error) {
+	// Create a new category
+	tagSpec := tags.Category{Name: categoryName, Cardinality: cardinality}
+	categoryID, err := tagManager.CreateCategory(ctx, &tagSpec)
+	if err != nil {
+		return "", fmt.Errorf("Error creating a new tag category %s, err: %v\n",
+			categoryName, err)
+	}
+	fmt.Printf("Successfully created a new category %s with categoryID %v\n",
+		categoryName, categoryID)
+
+	return categoryID, nil
+}
+
+// CreateNewTag creates a new tag within the given category
+func CreateNewTag(ctx context.Context, categoryId string, tagName string,
+	description string, tagManager *tags.Manager) (string, error) {
+	// Create a new tag within the given category
+	tagID, err := tagManager.CreateTag(ctx, &tags.Tag{Name: tagName, CategoryID: categoryId,
+		Description: description})
+	if err != nil {
+		return "", fmt.Errorf("Error creating a new tag %s within category %s, err: %v\n",
+			tagName, categoryId, err)
+	}
+	fmt.Printf("Successfully created a new tag %s with tagID %v within category %s\n",
+		tagName, tagID, categoryId)
+
+	return tagID, nil
+}
+
+// AttachTag attaches a tag to the given managed object
+func AttachTag(ctx context.Context, tagID string, moref types.ManagedObjectReference,
+	tagManager *tags.Manager) error {
+	err := tagManager.AttachTag(ctx, tagID, moref)
+	if err != nil {
+		return fmt.Errorf("Error attaching a tag %s to moref %v, err: %v",
+			tagID, moref, err)
+	}
+	fmt.Printf("Successfully attached a tag %s with moref %v\n", tagID, moref)
+
+	return nil
+}
+
+// TestGetZoneRegionOfVM creates new category and tag using tag manager for region
+// and zone respectively. It attaches region tag to datacenter and zone tag to cluster moref.
+// After that it calls existing function GetZoneRegion for one of the VMs belonging to this datacenter
+// and cluster and checks if we can successfully fetch tags from the ancestors of this VM that we added
+// using tagManager.
+func TestGetZoneRegionOfVM(t *testing.T) {
```

Comment:
Can we break this unit test into two tests?

"TestGetZoneRegion" and "TestIsInZoneRegion"?

Reply:
Done